tests/test_connection.c

```diff
@@ -548,6 +553,77 @@ static int s_test_connection_h2_prior_knowledge_not_work_with_tls(struct aws_all
 }
 AWS_TEST_CASE(connection_h2_prior_knowledge_not_work_with_tls, s_test_connection_h2_prior_knowledge_not_work_with_tls);
 
+static void s_on_tester_negotiation_result(
+    struct aws_channel_handler *handler,
+    struct aws_channel_slot *slot,
+    int err_code,
+    void *user_data) {
+
+    (void)slot;
+    (void)err_code;
+    struct tester *tester = (struct tester *)user_data;
+    struct aws_byte_buf src = aws_tls_handler_protocol(handler);
+    aws_byte_buf_init_copy(tester->negotiated_protocol, tester->alloc, &src);
+}
+
+static int s_test_connection_customized_alpn(struct aws_allocator *allocator, void *ctx) {
+    (void)ctx;
+#ifdef __APPLE__ /* Something is wrong with APPLE */
+    return AWS_OP_SUCCESS;
+#endif
+    char customized_alpn_string[] = "myh2";
```

Comment:
you should probably put the rest of this code in an `#else` block
won't the compiler complain about code that's impossible to reach after the `return`?

Reply:
Don't know... But, I'll remove this block after we have the Mac virtual env back to work. We can ignore it for now. (I can keep this PR open until then)